.github/workflows/build.yml

```diff
@@ -14,51 +14,27 @@ jobs:
   tests:
     strategy:
       fail-fast: false
-      max-parallel: 4
+      max-parallel: 5
       matrix:
         platform: [ubuntu-latest] #, windows-latest]
-        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
-        include:
-          - python-version: 3.8
-            tox-env: py38
-            continue-on-error: False
-          - python-version: 3.9
-            tox-env: py39
-            continue-on-error: False
-          - python-version: '3.10'
-            tox-env: py310
-            continue-on-error: False
-          - python-version: '3.11'
-            tox-env: py311
-            continue-on-error: False
-          - python-version: '3.12'
-            tox-env: py312
-            continue-on-error: False
+        python-version: [3.8, 3.9, '3.10', 3.11, 3.12]
         # exclude:
         #   - platform: windows-latest
         #     python-version: 3.9
 
-    env:
-      TOXENV: ${{ matrix.tox-env }}
-
     runs-on: ${{ matrix.platform }}
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - name: Set up Python ${{ matrix.python-version }}
-      if: "!endsWith(matrix.python-version, '-dev')"
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
         allow-prereleases: true
-    - name: Set up development Python ${{ matrix.python-version }}
-      if: endsWith(matrix.python-version, '-dev')
-      uses: deadsnakes/action@v2.1.1
-      with:
-        python-version: ${{ matrix.python-version }}
     - name: Python (Install Dependencies)
       run: |
-        python -m pip install --upgrade pip setuptools tox coverage build
+        python -m pip install --upgrade pip
```

Comment:
This seems like an unnecessary change. If any of these are installed, ensuring we have the latest is not a problem, if they are not installed, they will still be installed.

Reply:
There is no need to upgrade pip if you have already installed your dependencies on the old version.  See the GitHub Actions examples.

Reply:
My point generally was that I wanted to ensure all the listed packages were included as the latest, that includes `setuptools`, `tox`, `coverage`, `build`. Admittedly, some of these are never on the system, so they just install as normal, but some, like `setuptools` often is, and I want to ensure the latest is there.

That is why I don't understand why we want to only upgrade pip, just upgrade them all, and if they are already upgraded, they wont update.